cmd/podman/unpause.go

```diff
@@ -37,28 +52,56 @@ func unpauseCmd(c *cli.Context) error {
 	defer runtime.Shutdown(false)
 
 	args := c.Args()
-	if len(args) < 1 {
+	if len(args) < 1 && !c.Bool("all") {
 		return errors.Errorf("you must provide at least one container name or id")
 	}
-
-	var lastError error
-	for _, arg := range args {
-		ctr, err := runtime.LookupContainer(arg)
+	if c.Bool("all") {
+		cs, err := getAllOrLatestContainers(c, runtime, libpod.ContainerStatePaused, "paused")
 		if err != nil {
-			if lastError != nil {
-				fmt.Fprintln(os.Stderr, lastError)
+			return err
+		}
+		unpauseContainers = append(unpauseContainers, cs...)
+	} else {
+		for _, arg := range args {
+			ctr, err := runtime.LookupContainer(arg)
+			if err != nil {
+				return nil
 			}
-			lastError = errors.Wrapf(err, "error looking up container %q", arg)
-			continue
+			unpauseContainers = append(unpauseContainers, ctr)
 		}
-		if err = ctr.Unpause(); err != nil {
-			if lastError != nil {
-				fmt.Fprintln(os.Stderr, lastError)
+	}
+
+	// Assemble the unpause funcs
+	for _, ctr := range unpauseContainers {
+		con := ctr
+		f := func() error {
+			return con.Unpause()
```

Comment:
This won't work for `all` I think - it'll throw an error on every container that is not paused

Reply:
note that the unpauseContainers are containers in the state of paused ... so we should be ok

Reply:
Could still be racy - but probably fine, given that we guarantee they are paused.

Reply:
Wait, where are you verifying this? I don't see any code to filter the containers in question

Reply:
... discussed in irc ...